src/tests/rnp_tests.c

```diff
@@ -156,13 +145,132 @@ main(int argc, char *argv[])
+                void *state;
+                if (setup_test(&state)) {
+                    continue;
+                }
+                tests[j].test_func(&state);
+                teardown_test(&state);
+            }
         }
+        ret = 0;
     }
+
+    return ret;
+}
+
+int
+run_cli()
+{
+    int ret, i;
+
+    struct CMUnitTest clitests[] = {cmocka_unit_test(cli_rnp), cmocka_unit_test(cli_rnpkeys)};
+
+    for (i = 0; i < sizeof(clitests) / sizeof(clitests[0]); i++) {
+        clitests[i].setup_func = setup_test;
+        clitests[i].teardown_func = teardown_test;
+    }
+
+    ret = cmocka_run_group_tests(clitests, setup_test_group, teardown_test_group);
+    return ret;
+}
+
+int
+run_perf()
+{
+    int ret, i;
+
+    struct CMUnitTest perftests[] = {cmocka_unit_test(cli_performance)};
+
+    for (i = 0; i < sizeof(perftests) / sizeof(perftests[0]); i++) {
+        perftests[i].setup_func = setup_test;
+        perftests[i].teardown_func = teardown_test;
+    }
+
+    ret = cmocka_run_group_tests(perftests, setup_test_group, teardown_test_group);
+    return ret;
+}
+
+void
+print_usage()
+{
+    printf("Usage: rnp_tests [cli | perf | lib]\nWithout parameters lib tests are selected\n");
+}
+
+int
+main(int argc, char **argv)
+{
+    int  ret;
+    int  tests = TST_LIB;
+    char cwd[PATH_MAX];
+
+    exe_path = argv[0];
+    assert_non_null(getcwd(original_dir, sizeof(original_dir)));
+
+    /* Saving original working directory so we can use it later while calling python scripts.
+     * Currently they work only if called from the original directory.
+     * This was done before exe_path was introduced in parallel branches, needs review
+     */
+    assert_non_null(getcwd(cwd, sizeof(cwd)));
+    assert_int_equal(0, setenv("ORIGCWD", cwd, 1));
+
+    /* Create a temporary HOME.
+     * This is just an extra guard to protect against accidental
+     * modifications of a user's HOME.
+     */
+    char *tmphome = make_temp_dir();
+    assert_int_equal(0, setenv("HOME", tmphome, 1));
+    assert_int_equal(0, chdir(tmphome));
+
+    /* We use LOGNAME in a few places within the tests
+     * and it isn't always set in every environment.
+     */
+    if (!getenv("LOGNAME"))
+        setenv("LOGNAME", "test-user", 1);
+
+    int iterations = 0;
+    if (getenv("RNP_TEST_ITERATIONS")) {
+        iterations = atoi(getenv("RNP_TEST_ITERATIONS"));
+    }
+
+    /* Checking input parameters */
+    if (argc == 2) {
+        if (strcmp(argv[1], "cli") == 0) {
+            tests = TST_CLI;
+        } else if (strcmp(argv[1], "perf") == 0) {
+            tests = TST_PERF;
+        } else if (strcmp(argv[1], "lib") == 0) {
+            tests = TST_LIB;
+        } else {
+            printf("Wrong parameter: %s\n", argv[1]);
+            print_usage();
+            return 1;
+        }
+    } else if (argc > 2) {
+        printf("Wrong parameter count.\n");
+        print_usage();
+        return 1;
+    }
+
```

Comment:
I think you should add way to run `cli` tests over travis-ci. Because right now it runs only `lib` test.